cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -206,20 +209,50 @@ func NewCommand() *cobra.Command {
 }
 
 func getClusterFilter(kubeClient *kubernetes.Clientset, settingsMgr *settings.SettingsManager, shardingAlgorithm string) sharding.ClusterFilterFunction {
-	replicas := env.ParseNumFromEnv(common.EnvControllerReplicas, 0, 0, math.MaxInt32)
+
+	var replicas int
 	shard := env.ParseNumFromEnv(common.EnvControllerShard, -1, -math.MaxInt32, math.MaxInt32)
+
+	appControllerDeployment, _ := kubeClient.AppsV1().Deployments(settingsMgr.GetNamespace()).Get(context.Background(), common.ApplicationController, metav1.GetOptions{})
```

Comment:
This seems a bit fragile as if someone changes the application controller name, it would break the rebalancing feature. Maybe we shouldn't base on the name but on a predefined label? Or maybe something else to make it more robust? WDYT?

Reply:
A very valid point. The same is true for a predefined label value. As discussed over slack, I will add an e2e test to make sure we validate on the controller name or fail the test, thus failing the build.